airflow/gcp/hooks/pubsub.py

```diff
@@ -177,40 +293,88 @@ def create_subscription(
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: Sequence[Tuple[str, str]]]
         :return: subscription name which will be the system-generated value if
             the ``subscription`` parameter is not supplied
         :rtype: str
         """
-        service = self.get_conn()
-        full_topic = _format_topic(topic_project, topic)
+        subscriber = self.subscriber_client
+
         if not subscription:
             subscription = 'sub-{}'.format(uuid4())
         if not subscription_project:
             subscription_project = topic_project
-        full_subscription = _format_subscription(subscription_project,
-                                                 subscription)
-        body = {
-            'topic': full_topic,
-            'ackDeadlineSeconds': ack_deadline_secs
-        }
+
+        # Add airflow-version label to the subscription
+        labels = labels or {}
+        labels['airflow-version'] = 'v' + version.replace('.', '-').replace('+', '-')
+
+        # pylint: disable=no-member
+        subscription_path = SubscriberClient.subscription_path(subscription_project, subscription)
+        topic_path = SubscriberClient.topic_path(topic_project, topic)
+
+        self.log.info("Creating subscription (path) %s for topic (path) %a", subscription_path, topic_path)
         try:
-            service.projects().subscriptions().create(  # pylint: disable=no-member
-                name=full_subscription, body=body).execute(num_retries=self.num_retries)
-        except HttpError as e:
-            # Status code 409 indicates that the subscription already exists.
-            if str(e.resp['status']) == '409':
-                message = 'Subscription already exists: {}'.format(
-                    full_subscription)
-                self.log.warning(message)
-                if fail_if_exists:
-                    raise PubSubException(message)
-            else:
-                raise PubSubException(
-                    'Error creating subscription {}'.format(full_subscription),
-                    e)
+            subscriber.create_subscription(
+                name=subscription_path,
+                topic=topic_path,
+                push_config=push_config,
+                ack_deadline_seconds=ack_deadline_secs,
+                retain_acked_messages=retain_acked_messages,
+                message_retention_duration=message_retention_duration,
+                labels=labels,
+                retry=retry,
+                timeout=timeout,
+                metadata=metadata,
+            )
+        except AlreadyExists:
+            message = 'Subscription already exists: {}'.format(subscription_path)
+            self.log.warning(message)
```

Comment:
We should avoid formatting string before passing to logger. 
Reference
https://github.com/apache/airflow/pull/4804/files
https://github.com/apache/airflow/pulls?utf8=%E2%9C%93&q=is%3Apr+author%3Anuclearpinguin+colour

Reply:
Fixed.